specifyweb/frontend/js_src/lib/components/Atoms/Form.tsx

```diff
@@ -155,6 +155,7 @@ export const Input = {
         props.onChange?.(event);
       },
       readOnly: isReadOnly,
+      disabled: isReadOnly,
```

Comment:
note, readOnly and disabled are not equivalent
this change affects all checkboxes across specify, and that is likely not desirable in all cases
read more about the difference: https://ux.stackexchange.com/a/118041/131886
for example, readonly checkboxes are tab-focusable, but disabled are not.

Instead, notice that on line 153 in this file we are ignoring the change event for any checkbox that is disabled or read-only. Could you check why does that not prevent the change event?

<details><summary>Soluton</summary>
<p>

The checkbox in the but report has `onClick` event listener rather than `onChange` or `onValueChange`.
Could you modify the typings for `Input.Checkbox` to disallow using the onClick event, OR, add similar logic we have for onChange (disabled and readonly check) to the onClcik event

</p>
</details>